test/format/verify.c

```diff
@@ -138,42 +138,56 @@ table_mirror_fail_msg_flcs(WT_SESSION *session, const char *checkpoint, TABLE *b
 }
 
 /*
- * position_cursor_before --
- *     Place a cursor on the key directly preceding the target key.
+ * position_cursor_ge --
+ *     Place a cursor on the key greater than or equal to the target key.
  */
-static void
-position_cursor_before(TABLE *table, WT_CURSOR *cursor, uint64_t target_keyno)
+static int
+position_cursor_ge(
+  TABLE *table, WT_CURSOR *cursor, uint64_t target_keyno, WT_ITEM *key, uint64_t *keynop)
 {
     WT_DECL_RET;
-    WT_ITEM key;
-    int exact;
-
-    key_gen_init(&key);
-
-    switch (table->type) {
-    case FIX:
-    case VAR:
-        cursor->set_key(cursor, target_keyno);
-        break;
-    case ROW:
-        key_gen(table, &key, target_keyno);
-        cursor->set_key(cursor, &key);
-        break;
-    }
+    uint64_t table_keyno;
+    uint8_t table_bitv;
 
-    testutil_check(read_op(cursor, SEARCH_NEAR, &exact));
+    table_keyno = 0;
 
-    /* If we're on or past the target key then move backwards one key. */
-    if (exact >= 0) {
-        ret = read_op(cursor, PREV, NULL);
-        /*
-         * WT_NOTFOUND is ok here since any subsequent cursor->next calls will start from the
-         * beginning of the table.
-         */
-        testutil_assert(ret == 0 || ret == WT_NOTFOUND);
-    }
+    do {
+        switch (table->type) {
+        case FIX:
+            /*
+             * RS and VLCS skip over removed entries, FLCS returns a value of 0. Skip to the next
+             * matching key number or the next nonzero value. If the latter comes early, we'll visit
+             * the mismatch logic below.
+             */
+            for (;;) {
+                ret = read_op(cursor, NEXT, NULL);
+                testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+                if (ret != 0)
+                    break;
+                testutil_check(cursor->get_key(cursor, &table_keyno));
+                if (table_keyno >= target_keyno || table_keyno > TV(RUNS_ROWS))
+                    break;
+                testutil_check(cursor->get_value(cursor, &table_bitv));
+                if (table_bitv != 0)
+                    break;
+            }
+            break;
+        case VAR:
+            ret = read_op(cursor, NEXT, NULL);
+            testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+            if (ret == 0)
+                testutil_check(cursor->get_key(cursor, &table_keyno));
+            break;
+        case ROW:
+            ret = table_mirror_row_next(table, cursor, key, &table_keyno);
```

Comment:
@sueloverso My current solution is to place the cursor at the position that is larger than or equal to the target key but it leads to data mismatch. I did some debugging. It turns out we sometimes overshoot the target. For example, if the target key number is 100, we get 120 for the base cursor and 100 for the table cursor. It's not clear to me what's the difference of my change to the original change.

If the key 100 is deleted from the base table concurrently, that would explain what we see here. But does the old version have the same problem?